# Use KeyStore.getDefaultType() instead of hardcoded "JKS" for keystore defaults.
# This allows BCFKS (BouncyCastle FIPS KeyStore) when configured via java.security.
Index: kafka/clients/src/main/java/org/apache/kafka/common/config/SslConfigs.java
===================================================================
--- kafka.orig/clients/src/main/java/org/apache/kafka/common/config/SslConfigs.java
+++ kafka/clients/src/main/java/org/apache/kafka/common/config/SslConfigs.java
@@ -18,6 +18,7 @@ package org.apache.kafka.common.config;
 
 import org.apache.kafka.common.config.internals.BrokerSecurityConfigs;
 
+import java.security.KeyStore;
 import java.util.List;
 import java.util.Set;
 
@@ -58,7 +59,7 @@ public class SslConfigs {
     public static final String SSL_KEYSTORE_TYPE_CONFIG = "ssl.keystore.type";
     public static final String SSL_KEYSTORE_TYPE_DOC = "The file format of the key store file. "
             + "This is optional for client. The values currently supported by the default <code>ssl.engine.factory.class</code> are [JKS, PKCS12, PEM].";
-    public static final String DEFAULT_SSL_KEYSTORE_TYPE = "JKS";
+    public static final String DEFAULT_SSL_KEYSTORE_TYPE = KeyStore.getDefaultType();
 
     public static final String SSL_KEYSTORE_KEY_CONFIG = "ssl.keystore.key";
     public static final String SSL_KEYSTORE_KEY_DOC = "Private key in the format specified by 'ssl.keystore.type'. "
@@ -88,7 +89,7 @@ public class SslConfigs {
 
     public static final String SSL_TRUSTSTORE_TYPE_CONFIG = "ssl.truststore.type";
     public static final String SSL_TRUSTSTORE_TYPE_DOC = "The file format of the trust store file. The values currently supported by the default <code>ssl.engine.factory.class</code> are [JKS, PKCS12, PEM].";
-    public static final String DEFAULT_SSL_TRUSTSTORE_TYPE = "JKS";
+    public static final String DEFAULT_SSL_TRUSTSTORE_TYPE = KeyStore.getDefaultType();
 
     public static final String SSL_TRUSTSTORE_LOCATION_CONFIG = "ssl.truststore.location";
     public static final String SSL_TRUSTSTORE_LOCATION_DOC = "The location of the trust store file.";
